apps/src/Register3D.cpp

```diff
@@ -0,0 +1,185 @@
+#include <fstream>
+#include <iostream>
+
+#include <boost/filesystem.hpp>
+#include <itkCompositeTransform.h>
+#include <itkTransformFileReader.h>
+#include <itkTransformFileWriter.h>
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <vc/core/io/OBJReader.hpp>
+#include <vc/core/io/OBJWriter.hpp>
+#include <vc/core/types/Exceptions.hpp>
+#include <vc/core/types/UVMap.hpp>
+
+#include "rt/DeformableRegistration.hpp"
+#include "rt/ImageTransformResampler.hpp"
+#include "rt/ImageTypes.hpp"
+#include "rt/LandmarkIO.hpp"
+#include "rt/LandmarkRegistration.hpp"
+#include "rt/itk/itkOpenCVImageBridge.h"
+
+namespace fs = boost::filesystem;
+
+// Composite Transform
+using CompositeTransform = itk::CompositeTransform<double, 2>;
+
+// IO
+using TransformWriter = itk::TransformFileWriterTemplate<double>;
+using TransformReader = itk::TransformFileReaderTemplate<double>;
+
+int main(int argc, char* argv[])
+{
+    if (argc < 7) {
+        std::cerr << "Missing Parameters " << std::endl;
+        std::cerr << "Usage: " << argv[0];
+        std::cerr << " objFile landmarksFile ";
+        std::cerr << "movingImage iterations";
+        std::cerr << "objOutputFileName ";
+        std::cerr << "transformOutputFileName " << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    auto objFileName = argv[1];
+    auto landmarksFileName = argv[2];
+    // auto transformIn = argv[2];
+    auto movingImageFileName = argv[3];
+    auto iterationsIn = argv[4];
+    auto objOut = argv[5];
+    auto transformOut = argv[6];
+
+    printf("%-17s\n\n", "Registration Parameters");
+    printf("%-17s %s\n", "Obj file: ", objFileName);
+    printf("%-17s %s\n", "Landmarks file: ", landmarksFileName);
+    printf("%-17s %s\n", "Moving image: ", movingImageFileName);
+    printf("%-17s %s\n", "Iterations: ", iterationsIn);
+
+    ///// Setup input files /////
+    rt::Image8UC3::Pointer fixedImage;
+    rt::Image8UC3::Pointer movingImage;
+
+    // Read the OBJ file
+    volcart::io::OBJReader reader;
+    reader.setPath(objFileName);
+    volcart::ITKMesh::Pointer origMesh;
+    cv::Mat imageFixed;
+    try {
+        origMesh = reader.read();
+        imageFixed = reader.getTextureMat();
+        fixedImage =
+            itk::OpenCVImageBridge::CVMatToITKImage<rt::Image8UC3>(imageFixed);
+    } catch (volcart::IOException& excp) {
+        std::cerr << "Exception thrown " << std::endl;
+        std::cerr << excp.what() << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    auto cvMovingImage = cv::imread(movingImageFileName);
+    // Read the two images
```

Comment:
Move this comment up above the previous line and change what it says to match what's happening.